Source/JavaScriptCore/assembler/MacroAssemblerARMv7.h

```diff
@@ -2554,9 +2554,11 @@ class MacroAssemblerARMv7 : public AbstractMacroAssembler<Assembler> {
     Jump branch32(RelationalCondition cond, RegisterID left, RegisterID right)
     {
         if (left == ARMRegisters::sp) {
+            ASSERT(right != addressTempRegister);
```

Comment:
Why not use dataTempRegister if it conflicts?

Reply:
We have another patch trying to remove it

Reply:
Basically, we need another free register to use to make dataIC work, so the plan is to slowly free up one extra masm register.